src/components/OptionsSelector/BaseOptionsSelector.js

```diff
@@ -61,620 +53,652 @@ const defaultProps = {
     ...optionsSelectorDefaultProps,
 };
 
-class BaseOptionsSelector extends Component {
-    constructor(props) {
-        super(props);
-
-        this.updateFocusedIndex = this.updateFocusedIndex.bind(this);
-        this.scrollToIndex = this.scrollToIndex.bind(this);
-        this.selectRow = this.selectRow.bind(this);
-        this.selectFocusedOption = this.selectFocusedOption.bind(this);
-        this.addToSelection = this.addToSelection.bind(this);
-        this.updateSearchValue = this.updateSearchValue.bind(this);
-        this.incrementPage = this.incrementPage.bind(this);
-        this.sliceSections = this.sliceSections.bind(this);
-        this.calculateAllVisibleOptionsCount = this.calculateAllVisibleOptionsCount.bind(this);
-        this.handleFocusIn = this.handleFocusIn.bind(this);
-        this.handleFocusOut = this.handleFocusOut.bind(this);
-        this.debouncedUpdateSearchValue = _.debounce(this.updateSearchValue, CONST.TIMING.SEARCH_OPTION_LIST_DEBOUNCE_TIME);
-        this.relatedTarget = null;
-        this.accessibilityRoles = _.values(CONST.ROLE);
-        this.isWebOrDesktop = [CONST.PLATFORM.DESKTOP, CONST.PLATFORM.WEB].includes(getPlatform());
-
-        const allOptions = this.flattenSections();
-        const sections = this.sliceSections();
-        const focusedIndex = this.getInitiallyFocusedIndex(allOptions);
-        this.focusedOption = allOptions[focusedIndex];
-
-        this.state = {
-            sections,
-            allOptions,
-            focusedIndex,
-            shouldDisableRowSelection: false,
-            errorMessage: '',
-            paginationPage: 1,
-            disableEnterShortCut: false,
-            value: '',
-        };
-    }
-
-    componentDidMount() {
-        this.subscribeToEnterShortcut();
-        this.subscribeToCtrlEnterShortcut();
-        this.subscribeActiveElement();
-
-        if (this.props.isFocused && this.props.autoFocus && this.textInput) {
-            this.focusTimeout = setTimeout(() => {
-                this.textInput.focus();
-            }, CONST.ANIMATED_TRANSITION);
-        }
-
-        this.scrollToIndex(this.props.selectedOptions.length ? 0 : this.state.focusedIndex, false);
-    }
-
-    componentDidUpdate(prevProps, prevState) {
-        if (prevState.disableEnterShortCut !== this.state.disableEnterShortCut) {
-            // Unregister the shortcut before registering a new one to avoid lingering shortcut listener
-            this.unsubscribeEnter();
-            if (!this.state.disableEnterShortCut) {
-                this.subscribeToEnterShortcut();
+function BaseOptionsSelector(props) {
+    const isFocused = useIsFocused();
+    const {translate} = useLocalize();
+    const themeStyles = useThemeStyles();
+
+    const getInitiallyFocusedIndex = useCallback(
```

Comment:
In my experience, if you're using `useCallback` for a generator function that's only used once, you probably should use `useMemo` instead. I think that results in simpler code:

```diff
diff --git a/src/components/OptionsSelector/BaseOptionsSelector.js b/src/components/OptionsSelector/BaseOptionsSelector.js
index 23451b8e1a..a182753387 100755
--- a/src/components/OptionsSelector/BaseOptionsSelector.js
+++ b/src/components/OptionsSelector/BaseOptionsSelector.js
@@ -58,27 +58,6 @@ function BaseOptionsSelector(props) {
     const {translate} = useLocalize();
     const themeStyles = useThemeStyles();
 
-    const getInitiallyFocusedIndex = useCallback(
-        (allOptions) => {
-            let defaultIndex;
-            if (props.shouldTextInputAppearBelowOptions) {
-                defaultIndex = allOptions.length;
-            } else if (props.focusedIndex >= 0) {
-                defaultIndex = props.focusedIndex;
-            } else {
-                defaultIndex = props.selectedOptions.length;
-            }
-            if (_.isUndefined(props.initiallyFocusedOptionKey)) {
-                return defaultIndex;
-            }
-
-            const indexOfInitiallyFocusedOption = _.findIndex(allOptions, (option) => option.keyForList === props.initiallyFocusedOptionKey);
-
-            return indexOfInitiallyFocusedOption;
-        },
-        [props.shouldTextInputAppearBelowOptions, props.initiallyFocusedOptionKey, props.selectedOptions.length, props.focusedIndex],
-    );
-
     const isWebOrDesktop = [CONST.PLATFORM.DESKTOP, CONST.PLATFORM.WEB].includes(getPlatform());
     const accessibilityRoles = _.values(CONST.ROLE);
 
@@ -137,7 +116,24 @@ function BaseOptionsSelector(props) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
     const initialAllOptions = useMemo(() => flattenSections(), []);
     const [allOptions, setAllOptions] = useState(initialAllOptions);
-    const [focusedIndex, setFocusedIndex] = useState(getInitiallyFocusedIndex(initialAllOptions));
+
+    const initialFocusedIndex = useMemo(() => {
+        if (!_.isUndefined(props.initiallyFocusedOptionKey)) {
+            return _.findIndex(allOptions, (option) => option.keyForList === props.initiallyFocusedOptionKey);
+        }
+
+        let defaultIndex;
+        if (props.shouldTextInputAppearBelowOptions) {
+            defaultIndex = allOptions.length;
+        } else if (props.focusedIndex >= 0) {
+            defaultIndex = props.focusedIndex;
+        } else {
+            defaultIndex = props.selectedOptions.length;
+        }
+        return defaultIndex;
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- this value is only used to initialize state so only ever needs to be computed on the first render
+    }, []);
+    const [focusedIndex, setFocusedIndex] = useState(initialFocusedIndex);
     const [focusedOption, setFocusedOption] = useState(allOptions[focusedIndex]);
 
     /**

```